src/main/java/io/cdap/e2e/pages/locators/WranglerLocators.java

```diff
@@ -0,0 +1,202 @@
+    "-decimal-applyButton']")
+  public static WebElement applyButtonUppercase;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-defineVariable-" +
+    "conditionTextInput']")
+  public static WebElement variableValue;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-parse-modal-excel-" +
+    "sheetNumberInput']")
+  public static WebElement excelSheetNumber;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-findAndReplace-oldValueInput']")
+  public static WebElement enterOldValue;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-testid='features-dataprep-directives-encodeDecode-title-Encode']")
+  public static WebElement encodeDirective;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-testid='features-dataprep-directives-encodeDecode-title-Decode']")
+  public static WebElement decodeDirective;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-findAndReplace-newValueInput']")
+  public static WebElement enterNewValue;
+
+  @FindBy(how = How.XPATH, using = "button[@data-testid='features-dataprep-directives-extractFields-modal-patterns-" +
+    "patternsSelector']")
+  public static WebElement selectPatternButton;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-parse-modal-singleField-" +
+    "mainFieldInput']")
+  public static WebElement columnWidths;
+
+  @FindBy(how = How.XPATH, using = "//select[@data-testid='features-dataprep-directives-defineVariable-" +
+    "columnSelector']")
+  public static WebElement selectColumn;
+
+  @FindBy(how = How.XPATH, using = "//select[@data-testid='features-dataprep-directives-defineVariable-" +
+    "conditionSelector']")
+  public static WebElement selectRow;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-testid='features-dataprep-directives-parse-modal-" +
+    "singleField-mainFieldInput']")
+  public static WebElement enterDepthJson;
+
+  @FindBy(how = How.XPATH, using = "//*[@data-testid='features-dataprep-directives-parse-modal-xmlToJson-depthInput']")
+  public static WebElement enterDepthXmlToJson;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-parse-modal-dateFormats-" +
+    "customFormatInput']")
+  public static WebElement customDate;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-setCounter-counterNameInput']")
+  public static WebElement enterCounterName;
+
+  @FindBy(how = How.XPATH, using = "//input[@data-testid='features-dataprep-directives-setCounter-" +
+    "counterIncrementInput']")
+  public static WebElement incrementCount;
+
+  public static WebElement locateTransformationButton(String columnName) {
+    return SeleniumDriver.getDriver()
+      .findElement(By.xpath("//th[@id='column-" + columnName + "']//button[@data-testid='" +
```

Comment:
Is the `//th[@id= ... ]` necessary ? Can the column header not be identified using `data-testid` ?

Reply:
No , we want to give the column name as well for eg =**city or name** and that cannot be identified using data-testid